### PR TITLE
add --dirty flag to skip download/extraction step

### PR DESCRIPTION
**Pull request by msarahan**
I've been working on Qt5 builds.  Extracting the source takes 5-10 *minutes*.  This PR avoids it.  This is a gun pointed at your foot - use it to iterate quickly on recipes, but keep in mind that it may bite you if you forget to capture any source code changes as patches or changes to your recipe.

### CODE REVIEW

**Comment by patricksnape** [comment]
This would be extremely useful.

**Comment by msarahan** [comment]
Any other thoughts on improving this, @patricksnape?  If not, I'll merge this soon.

File: conda_build/source.py [hunk 1]
```diff
@@ -346,43 +346,45 @@ def apply_patch(src_dir, path):
...
+                move_to_trash(WORK_DIR, '')
+        else:
+            rm_rf(WORK_DIR)
+
+    if not os.path.exists(WORK_DIR):
+        if any(k in meta for k in ('fn', 'url')):
+            unpack(meta, verbose=verbose)
+        elif 'git_url' in meta:
+            git_source(meta, recipe_dir, verbose=verbose)
+        # build to make sure we have a work directory with source in it.  We want to make sure that
```
Comment (patricksnape) [inline conda_build/source.py]: The comments here are repeated

**Comment by patricksnape** [comment]
Do we really want to keep the `_build` environment around? I can see maintaining the `work` directory but generally creating the `_build` environment is usually fairly quick? I can foresee the `_build` environment potentially becoming quite 'dirty' if say files are being installed into the wrong places.

File: conda_build/source.py (hunk 1 above)
Comment (msarahan) [inline conda_build/source.py]: hah!  That's a weird copy/paste.  Thanks for pointing it out.

**Comment by msarahan** [comment]
Thanks @patricksnape.  I agree, and have integrated your suggestions.

**Pull request merged**